Editor/Window/GameLiftRequestAdapter.cs

```diff
@@ -0,0 +1,242 @@
+            {
+                var listLocationsResponse = await _gameLiftConfig.GameLiftWrapper.ListLocations(new ListLocationsRequest
+                {
+                    Filters = new List<string> { "CUSTOM" }
+                });
+
+                var foundLocation =
+                    listLocationsResponse.Locations.FirstOrDefault(l => l.LocationName.ToString() == fleetLocation);
+
+                if (foundLocation == null)
+                {
+                    var createLocationResponse = await _gameLiftConfig.GameLiftWrapper.CreateLocation(
+                        new CreateLocationRequest()
+                        {
+                            LocationName = fleetLocation
+                        });
+
+                    if (createLocationResponse.HttpStatusCode == HttpStatusCode.OK)
+                    {
+                        Debug.Log($"Created Custom Location {fleetLocation}");
+                    }
+                }
+
+                return true;
+            }
+            catch (Exception ex)
+            {
+                var errorBox = _container.Q<VisualElement>("FleetErrorInfoBox");
+                errorBox.style.display = DisplayStyle.Flex;
+                errorBox.Q<Label>().text = ex.Message;
+                return false;
+            }
+        }
+        
+        private async Task<string> CreateFleet(ComputeType computeType, string fleetLocation, string fleetName)
+        {
+            try
+            {
+                var createFleetRequest = new CreateFleetRequest
+                {
+                    Name = fleetName,
+                    ComputeType = computeType,
+                    Description = FleetDescription,
+                    Locations = new List<LocationConfiguration>
+                    {
+                        new()
+                        {
+                            Location = fleetLocation
+                        }
+                    },
+                    Tags = new List<Tag>
+                    {
+                        new()
+                        {
+                            Key = "CreatedBy",
+                            Value = "AmazonGameLiftUnityPlugin"
+                        }
+                    }
+                    };
+                var createFleetResponse = await _gameLiftConfig.GameLiftWrapper.CreateFleet(createFleetRequest);
+                
+                if (createFleetResponse.HttpStatusCode == HttpStatusCode.OK)
+                {
+                    Debug.Log($"Created Fleet {fleetName}");
+                }
+                
+                return createFleetResponse.FleetAttributes.FleetId;
+            }
+            catch (Exception ex)
+            {
+                var errorBox = _container.Q<VisualElement>("FleetErrorInfoBox");
+                errorBox.style.display = DisplayStyle.Flex;
+                errorBox.Q<Label>().text = ex.Message;
+                Debug.Log(ex.Message);
+                return null;
```

Comment:
Looks like common error handling for all cases. Can you factor this out into a util like 


```
private void HandleException(string elementName, string exceptionMsg) {
 ....
}
```

Reply:
refactored